### ISSUE TITLE
Sunrise sunset time in AM/PM format. How to change to 24HR format?

### ISSUE THREAD

**Issue by mmatesic01** [bramkragten/weather-card#3]
Hi,
How can I change for sunrise and sunset time to be in 24-HR format.
Currently I have AM/PM format.
My HA interface is in Slovenian language. Using DarkSky.

**Comment by fribse** [comment]
Mine is shown in 24-hr format, not sure if it's fixed, or it is just using the setting in HA?

**Comment by jokerigno** [comment]
Same "issue".

Italian language.

**Comment by disrupted** [comment]
same here

**Comment by kajhoej** [comment]
Same issue here. Log book and History in HA are all 24H times. Only this weather card seems to be showing in AM/PM

**Comment by Martorias** [comment]
Same here, can't find setting to change it.

**Comment by awolkers** [comment]
Same here, also the wind direction is shown in English while the weekdays are shown in Dutch. I set Dutch as my language for HA.

**Comment by exetico** [comment]
Same issue here, with AM/PM. 

What do to? :-)

**Comment by fohlsso2** [comment]
Not sure it is the fault of this card as I have seen the same issue in more places, but until it is fixed it is possible to locally do a workaround.

In the file weather-card.js the string toLocaleTimeString() are in two places, edit it to contain you selected country code, for me that is toLocaleTimeString('sv-SE'). You need to do the same to the file inside the archive (weather-card.js.gz)

**Comment by oz9els** [comment]
The AM/PM is a Home Assistant "feature" - It has to do with your browser/OS-setting. (try fiddling with it, hopefully it will work) Look at the original weather card - it also has the AM/PM ;-)
Is is on the bug-list/feature request list of Home Assistant.

**Comment by RezzZ** [comment]
> The AM/PM is a Home Assistant "feature" - It has to do with your browser/OS-setting. (try fiddling with it, hopefully it will work) Look at the original weather card - it also has the AM/PM ;-) Is is on the bug-list/feature request list of Home Assistant.

You sure its a feature? HA shows times without AM/PM in logbook and other places.

**Comment by awolkers** [comment]
It's a user setting in HA. Click on the user icon in the left pane to see all the settings related to formatting of date and time settings.

**Comment by RezzZ** [comment]
> It's a user setting in HA. Click on the user icon in the left pane to see all the settings related to formatting of date and time settings.

Settings are not honored
![image](https://github.com/bramkragten/weather-card/assets/70499/e26b9da2-784f-4052-ac2e-52778444a7b5)
![image](https://github.com/bramkragten/weather-card/assets/70499/9474eb20-9dac-42b5-a7b9-96cfaeeef4e2)

**Comment by MHBraun** [comment]
Same issue here. Any solution after 5 years?

**Comment by exetico** [comment]
@MHBraun I'm sure you're more than welcome to submit a PR.

A person working for free, doesn't speed up the work, cause you decide to make focus on "It's a long lasting issue" 👍.

**Comment by MHBraun** [comment]
Do not understand your comment or assumption. 
Can you please elaborate? 

I have the (similar) issue. 
Is there a solution for it? 
It is some time down the road, so there might be a fix which is not mentioned here. 

Markus

________________________________

**Comment by ralphbk** [comment]
Yes, it's in User Settings. If you have Language set to just "English" you'll get "AM" or "PM" suffixes. If you set it to "English (GB)" you'll get 24 hour time displayed.
